ufo/setup_test.py

```diff
@@ -0,0 +1,114 @@
+"""Test setup module functionality."""
+from mock import MagicMock
+from mock import patch
+
+import flask
+from googleapiclient import discovery
+import unittest
+
+from . import app
+import base_test
+from . import db
+import models
+import oauth
+import setup
+
+FAKE_OAUTH_URL = 'sftp://1800-oauth.com'
+FAKE_DOMAIN = 'yahoo.com'
+FAKE_OAUTH_CODE = 'foobar'
+
+class SetupTest(base_test.BaseTest):
+  """Test setup class functionality."""
+
+  def setUp(self):
+    """Setup test app on which to call handlers and db to query."""
+    super(SetupTest, self).setUp()
+    super(SetupTest, self).setup_config()
+
+  @patch('flask.render_template')
+  def testNotSetupHandler(self, mock_render_template):
+    """Test the not setup handler loads the error page as called."""
+    mock_render_template.return_value = ''
+
+    resp = self.client.get(flask.url_for('not_setup'))
+
+    args, kwargs = mock_render_template.call_args
+    self.assertEquals('error.html', args[0])
+    self.assertEquals(setup.PLEASE_CONFIGURE_TEXT, kwargs['error_text'])
+
+  @patch.object(oauth, 'getOauthFlow')
+  @patch('flask.render_template')
+  def testGetSetupHandler(self, mock_render_template, mock_oauth_flow):
+    """Test get on the setup handler loads the setup page."""
+    mock_render_template.return_value = ''
+    flow_object = mock_oauth_flow.return_value
+    flow_object.step1_get_authorize_url.return_value = FAKE_OAUTH_URL
+
+    resp = self.client.get(flask.url_for('setup'))
+
+    args, kwargs = mock_render_template.call_args
+    self.assertEquals('setup.html', args[0])
+    self.assertEquals(self.config, kwargs['config'])
+    self.assertEquals(FAKE_OAUTH_URL, kwargs['oauth_url'])
+
+  @patch.object(discovery, 'build')
+  @patch.object(oauth, 'getOauthFlow')
+  @patch('flask.render_template')
+  def testPostSetupIncorrectDomain(self, mock_render_template,
```

Comment:
nit: can we change the method name to be more easy to read, such as:
testPostSetupWithIncorrectDomain()